### DISCUSSION TITLE
how to remove unhandledPromiseRejectionWarning?

### DISCUSSION THREAD

**Discussion by Lancernix** [sidorares/node-mysql2#1538] [Q&A]
When I was testing whether the select function I wrote was correct, I deliberately wrote a wrong column name. Then the console output the corresponding error, but there was an `unhandledpromiserejectionwarning`. I added `try...catch` to all async functions, but warning still existed. It seems to be `promise.js` in mysql2 omits the handle of promise rejection? Or just there are something incorrect with my code? This warning does not affect the operation of the function, but how can I successfully remove the `unhandledpromiserejectionwarning`?

**code:**
![image](https://user-images.githubusercontent.com/26547236/147562011-713c1b79-5b0c-47f9-a12a-5e82764a260c.png)
![image](https://user-images.githubusercontent.com/26547236/147562064-9d1dc75a-c594-4582-8ca9-96d3461b0505.png)
![image](https://user-images.githubusercontent.com/26547236/147562091-92e924c7-f58d-4cfe-aa04-19793c2123e3.png)

**console:**

![image](https://user-images.githubusercontent.com/26547236/147562247-a6b04acd-3311-4fb1-a941-bba1407dae6b.png)
![image](https://user-images.githubusercontent.com/26547236/147562283-cfe7e976-ad3a-4cc3-aed5-e69c938af846.png)

**Comment by flipperlite** [comment]
Why would you want to remove it? It's an error that needs to be handled gracefully for the user. In your two functions, you have in your catch block `throw error;` which is correct IMHO. That means you're bubbling up the error and need to ultimately catch it at a higher level. If it goes unhandled, it may crash your app. For example if you were using NodeJS Express, you'd have a higher level try/catch block in your middleware function that would do a `return next(error);`. You could then have Express handle the error or use your own middleware function to scrub the raw error and display a user friendly error like "Oops. Something went wrong." so you're not sending out sensitive information. BTW, the try/catch block in your two functions are not necessary since you're not doing anything with the error object and you could remove the try/catch blocks assuming you catch the error upstream.

Here's a link on how to [handle errors in NodeJS Express](http://thecodebarbarian.com/80-20-guide-to-express-error-handling). If you're using another backend stack, the concepts are similar.

**Comment by sidorares** [comment]
@Lancernix I agree with @flipperlite comment - it's likely a sign of a problem in your code and need to be addressed as such ( adding an error handler, fixing actual error etc ). Closing the issue for now but feel free to post follow up questions here
